src/software/backend/simulation/simulator_robot.h

```diff
@@ -1,146 +1,106 @@
- * "instances" of robot firmware at once.
- */
-class SimulatorRobotSingleton
+class SimulatorRobot
 {
    public:
     /**
-     * Sets the ID of the robot being controlled by this class
+     * Create a new SimulatorRobot given a PhysicsRobot
      *
-     * @param id The ID of the robot to control
+     * @param physics_robot the PhysicsRobot to simulate and control
      */
-    static void setRobotId(unsigned int id);
+    explicit SimulatorRobot(std::weak_ptr<PhysicsRobot> physics_robot);
+    explicit SimulatorRobot() = delete;
 
     /**
-     * Sets the PhysicsRobots that can be controlled by this class
+     * Returns the ID of this robot
      *
-     * @param robots the PhysicsRobots that can be controlled by this class
+     * @return the ID of this robot
      */
-    static void setPhysicsRobots(const std::vector<std::weak_ptr<PhysicsRobot>>& robots);
+    unsigned int getRobotId();
 
-    /**
-     * Creates a FirmwareRobot corresponding to the current PhysicsRobot
-     *
-     * @return a FirmwareRobot corresponding to the current PhysicsRobot
-     */
-    static std::unique_ptr<FirmwareRobot_t, FirmwareRobotDeleter> createFirmwareRobot();
-
-   private:
     /**
      * Returns the x-position of the robot, in global field coordinates, in meters
      *
      * @return the x-position of the robot, in global field coordinates, in meters
      */
-    static float getPositionX();
+    float getPositionX();
 
     /**
      * Returns the y-position of the robot, in global field coordinates, in meters
      *
      * @return the y-position of the robot, in global field coordinates, in meters
      */
-    static float getPositionY();
+    float getPositionY();
+
+    /**
+     * Returns the current position of the robot, in global field coordinates, in meters
+     *
+     * @return the current position of the robot, in global field coordinates, in meters
+     */
+    Point position();
 
     /**
      * Returns the orientation of the robot, in global field coordinates, in radians
      *
      * @return the orientation of the robot, in global field coordinates, in radians
      */
-    static float getOrientation();
+    float getOrientation();
 
     /**
      * Returns the x-velocity of the robot, in global field coordinates, in m/s
      *
      * @return the x-velocity of the robot, in global field coordinates, in m/s
      */
-    static float getVelocityX();
+    float getVelocityX();
 
     /**
      * Returns the y-velocity of the robot, in global field coordinates, in m/s
      *
      * @return the y-velocity of the robot, in global field coordinates, in m/s
      */
-    static float getVelocityY();
+    float getVelocityY();
+
+    /**
+     * Returns the current linear velocity of the robot, in global field coordinates, in m
+     * / s
+     *
+     * @return the current linear velocity of the robot, in global field coordinates, in m
+     * / s
```

Comment:
looks like clang-format knackered the units a bit here

Reply:
:cry: fixed